contracts/programs/contract-reader-interface/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+use anchor_lang::prelude::*;
+use std::mem::size_of;
+
+declare_id!("6AfuXF6HapDUhQfE4nQG9C1SGtA1YjP3icaJyRfU4RyE");
+
+#[program]
+pub mod contract_reader_interface {
```

Comment:
nit: Should we call this `chain_components_...` since this contract will presumably also be used for ChainWriter?

Reply:
Do you mind if we make that change when we add `ChainWriter` in a followup PR?

Reply:
sure